### PR TITLE
Set MOUSE_FILTER_IGNORE for blocks where necessary

### PR DESCRIPTION
**Pull request by dylanmccall**
Certain components inside our Block scenes were consuming input events, making it difficult to interact with other blocks positioned around them.

### CODE REVIEW

File: addons/block_code/ui/blocks/utilities/snap_point/snap_point.tscn [hunk 1]
```diff
@@ -6,10 +6,9 @@
 anchors_preset = 15
 anchor_right = 1.0
 anchor_bottom = 1.0
-offset_right = -1152.0
-offset_bottom = -648.0
```
Comment (dbnicholson) [inline addons/block_code/ui/blocks/utilities/snap_point/snap_point.tscn:10]: This seems unrelated. Is it intentional?

File: addons/block_code/ui/blocks/statement_block/statement_block.tscn [hunk 2]
```diff
@@ -32,7 +34,6 @@ layout_mode = 2
 mouse_filter = 1
 script = ExtResource("2_lctqt")
 color = Color(1, 1, 1, 1)
-outline_color = Color(0.8, 0.8, 0.8, 1)
```
Comment (dbnicholson) [inline addons/block_code/ui/blocks/statement_block/statement_block.tscn:35]: Ack. I should have resaved this scene when removing the `outline_color` property in cc307af93c21f39bf695f205f8731385e703708a.

File: addons/block_code/ui/blocks/statement_block/statement_block.tscn [hunk 3]
```diff
@@ -54,6 +55,5 @@ theme_override_constants/separation = 0
 
 [node name="SnapPoint" parent="VBoxContainer" instance=ExtResource("3_5vaov")]
 layout_mode = 2
-block_path = NodePath("../..")
```
Comment (dbnicholson) [inline addons/block_code/ui/blocks/statement_block/statement_block.tscn:57]: Ack. The `block_path` removal is from f91ac7ee93fadda3fc26ca2f3f522aae7a008c53.

**Review by dbnicholson — COMMENTED** [review]
I trust this is a good thing to do, but I don't understand why the SnapPoint offset was removed.

File: addons/block_code/ui/blocks/utilities/snap_point/snap_point.tscn (hunk 1 above)
Comment (dylanmccall) [reply addons/block_code/ui/blocks/utilities/snap_point/snap_point.tscn:10]: Yeah, not intentional. Not sure what that's about - it doesn't really look _right_ - but no reason to change it here. I removed that change now :)

Comment (dbnicholson) [reply addons/block_code/ui/blocks/utilities/snap_point/snap_point.tscn:10]: Maybe it's something that was set in the scene at some point but actually had no effect? 🤷

**Review by dbnicholson — APPROVED** [review]
